src/Tribe/iCal.php

```diff
@@ -237,77 +321,287 @@ private function get_month_view_events() {
+		);
+	}
 
-		if ( $post ) {
-			$events_posts = is_array( $post ) ? $post : array( $post );
-		} elseif ( tribe_is_month() ) {
-			$events_posts = self::get_month_view_events();
-		} elseif ( tribe_is_organizer() ) {
-			$events_posts = $this->get_events_list( array(
-				'organizer'    => get_the_ID(),
-				'eventDisplay' => 'list',
-			) );
-		} elseif ( tribe_is_venue() ) {
-			$events_posts = $this->get_events_list( array(
-				'venue'        => get_the_ID(),
-				'eventDisplay' => tribe_get_request_var( 'tribe_event_display', 'list' ),
-			) );
-		} else {
+	/**
+	 * Get the file name of the *.ics file
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed The calendar name
+	 */
+	protected function get_file_name() {
+		$event_ids = wp_list_pluck( $this->events, 'ID' );
+		$site = sanitize_title( get_bloginfo( 'name' ) );
+		$hash = substr( md5( implode( $event_ids ) ), 0, 11 );
+		$filename = sprintf( '%s-%s.ics', $site, $hash );
 
-			if ( ! $wp_query = tribe_get_global_query_object() ) {
-				return;
+		/**
+		 * Modifies the filename provided in the Content-Disposition header for iCal feeds.
+		 *
+		 * @var string       $filename
+		 * @var WP_Post|null $post
+		 */
+		return apply_filters( 'tribe_events_ical_feed_filename', $filename, $this->post );
+	}
+
+	/**
+	 * Get the full content of the *.ics file.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	protected function get_content() {
+		$parts = [
+			$this->get_start(),
+			$this->get_timezones( $this->events ),
+			$this->get_body( $this->events ),
+			$this->get_end()
+		];
+		return implode( '', $parts );
+	}
+
+	/**
+	 * Get the start of the .ics File
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed
+	 */
+	protected function get_start() {
+		$blog_home    = get_bloginfo( 'url' );
+		$blog_name    = get_bloginfo( 'name' );
+
+		$content  = "BEGIN:VCALENDAR\r\n";
+		$content .= "VERSION:2.0\r\n";
+		$content .= 'PRODID:-//' . $blog_name . ' - ECPv' . Tribe__Events__Main::VERSION . "//NONSGML v1.0//EN\r\n";
+		$content .= "CALSCALE:GREGORIAN\r\n";
+		$content .= "METHOD:PUBLISH\r\n";
+
+		/**
+		 * Allows for customizing the value of the generated iCal file's "X-WR-CALNAME:" property.
+		 *
+		 * @param string $blog_name The value to use for "X-WR-CALNAME"; defaults to value of get_bloginfo( 'name' ).
+		 */
+		$x_wr_calname = apply_filters( 'tribe_ical_feed_calname', $blog_name );
+
+		if ( ! empty( $x_wr_calname ) ) {
+			$content .= 'X-WR-CALNAME:' . $x_wr_calname . "\r\n";
+		}
+
+		$content .= 'X-ORIGINAL-URL:' . $blog_home . "\r\n";
+		$content .= "X-WR-CALDESC:" . sprintf( esc_html_x( 'Events for %s', 'iCal feed description', 'the-events-calendar' ), $blog_name ) . "\r\n";
+
+		/**
+		 * Allows for customization of the various properties at the top of the generated iCal file.
+		 *
+		 * @param string $content Existing properties atop the file; starts at "BEGIN:VCALENDAR", ends at "X-WR-CALDESC".
+		 */
```

Comment:
Let's add a `@since TBD` here.

Reply:
Oh this one is an existing filter that was moved around due the changes on the code, do you think is still relevant to add a `@since` even was not added on this PR?

Reply:
Perfect!